packages/logger/src/formatters/omitFields.js

```diff
@@ -1,7 +1,11 @@
 import _ from 'lodash';
 
-export function omitFields(fieldsToOmit = []) {
+export function omitFields(fieldsToOmit = [], omitFieldsPathRoots = []) {
   return function omitFieldsLog(log) {
-    return _.omit(log, fieldsToOmit);
+    const result = _.omit(log, fieldsToOmit);
+    return omitFieldsPathRoots.reduce(
+      (acc, pathRoot) => (acc[pathRoot] ? { ...acc, [pathRoot]: _.omit(acc[pathRoot], fieldsToOmit) } : acc),
+      result,
+    );
```

Comment:
ok i think i got it now. it took me few mins to understand
you want to supply "another" root, in addition to the actual json for the omit to work on?
that's perry confusing and not intuitive.
let's talk about the issue first?
what are we trying to solve?
we want to remove fields from anywhere in the tree.
but you solution solves only for "known" roots, and it's not even dynamic?
i don't think it's worth the effort, as the solution is complex and not whole

Reply:
"we want to remove fields from anywhere in the tree."
- No, we don't want to remove fields from anywhere since we exactly know in advance possible structures of log.
Theoretically, we can think of dynamic behaviour, BUT it will take pretty much exec time for each log to process recursively, so currently, I think we follow KISS and address currently existing issue. 
Once we will see (with alert, which is ready) that we have other more complicated examples, we may make this logic more complex, maybe dynamic. According to the cases we have not, I don't want to overengineer it.

"but you solution solves only for "known" roots, and it's not even dynamic?"
- That's right, we don't need it now.

"as the solution is complex and not whole"
- Dynamic solution will be even more complex and may have performance impact.

Reply:
what about the most important one :)
"what is the problem that we are trying to solve"?

Reply:
We want to omit sensitive fields in logs

Reply:
we already can...
`logger.log(config)` will work
`logger.log({nested: config})` won't
so let's not nest........ WDYT? i think it's KISS

Reply:
Well, technically we can, I also thought about it, but we have a case (when remote config changed) when we really would like to log 2 configs in one log. Let's discuss it on today Whatsup Infra

Reply:
sure
but anyhow, im doubt it's that important... :)